matrix/lambdas/daemons/driver.py

```diff
@@ -1,35 +1,90 @@
-import itertools
-import math
 import typing
+import os
 
 import requests
 
-from matrix.common.aws.lambda_handler import LambdaHandler, LambdaName
-from matrix.common.aws.cloudwatch_handler import CloudwatchHandler, MetricName
+from matrix.common.config import MatrixInfraConfig, MatrixRedshiftConfig
 from matrix.common.logging import Logging
-from matrix.common.request.request_cache import RequestCache
 from matrix.common.request.request_tracker import RequestTracker, Subtask
+from matrix.common.aws.dynamo_handler import DynamoHandler
+from matrix.common.aws.sqs_handler import SQSHandler
+from matrix.common.aws.s3_handler import S3Handler
 
 logger = Logging.get_logger(__name__)
 
+expression_query_template = """
+    UNLOAD ($$SELECT cell.cellkey, expression.featurekey, expression.exrpvalue
+    FROM expression
+    LEFT OUTER JOIN feature on (expression.featurekey = feature.featurekey)
+    INNER JOIN cell on (expression.cellkey = cell.cellkey)
+    INNER JOIN analysis on (cell.analysiskey = analysis.analysiskey)
+    WHERE feature.isgene
+    AND expression.exprtype = 'Count'
+    AND analysis.bundle_fqid IN {3}$$)
+    TO 's3://{0}/{1}/expression'
+    IAM_ROLE '{2}'
+    GZIP
+    ALLOWOVERWRITE
+    MANIFEST VERBOSE
+    ;
+"""
+
+cell_query_template = """
+    UNLOAD($$SELECT cell.cellkey, cell.genes_detected, donor_organism.*, library_preparation.*, project.short_name
+    FROM cell
+    LEFT OUTER JOIN donor_organism on (cell.donorkey = donor_organism.donorkey)
+    LEFT OUTER JOIN library_preparation on (cell.librarykey = library_preparation.librarykey)
+    LEFT OUTER JOIN project on (cell.projectkey = project.projectkey)
+    INNER JOIN analysis on (cell.analysiskey = analysis.analysiskey)
+    WHERE analysis.bundle_fqid IN {3}$$)
+    TO 's3://{0}/{1}/cell_metadata'
+    IAM_ROLE '{2}'
+    GZIP
+    ALLOWOVERWRITE
+    MANIFEST VERBOSE
+    ;
+"""
+
+feature_query_template = """
+    UNLOAD ($$SELECT *
+    FROM feature
+    WHERE feature.isgene$$)
+    to 's3://{0}/{1}/gene_metadata'
+    IAM_ROLE '{2}'
+    GZIP
+    ALLOWOVERWRITE
+    MANIFEST VERBOSE;
+"""
```

Comment:
I think this is fine because it's a very fast query to run, but this is constant for every request.

Reply:
Yea we can optimize later